src/catalog/src/system_schema/pg_catalog.rs

```diff
@@ -69,30 +56,31 @@ impl SystemSchemaProvider for PGCatalogProvider {
     }
 }
 
-// TODO(j0hn50n133): Not sure whether to avoid duplication with `information_schema` or not.
-macro_rules! setup_memory_table {
-    ($name: expr) => {
-        paste! {
-            {
-                let (schema, columns) = get_schema_columns($name);
-                Some(Arc::new(MemoryTable::new(
-                    consts::[<PG_CATALOG_ $name  _TABLE_ID>],
-                    $name,
-                    schema,
-                    columns
-                )) as _)
-            }
-        }
-    };
-}
-
 impl PGCatalogProvider {
     pub fn new(catalog_name: String, catalog_manager: Weak<dyn CatalogManager>) -> Self {
+        let static_tables =
+            PgCatalogStaticTables::try_new().expect("Failed to initialize static tables");
+        let inner = PgCatalogSchemaProvider::try_new(
+            CatalogManagerWrapper {
+                catalog_name: catalog_name.clone(),
+                catalog_manager,
+            },
+            Arc::new(static_tables),
+        )
+        .expect("Failed to initialize PgCatalogSchemaProvider");
```

Comment:
Are these two expects safe?

Reply:
Yes, these errors are mostly loading schema from data file that bundled into the crate, which is a static behavior that can be verified by unit tests. 

But I will check again if we can use `try_new` here.

Reply:
Checked its usage, it doesn't allow us to return error. I think it's ok and safe for this.

Reply:
better add to safety comment